### PR TITLE
Move Updating of Crystallized State to Transition

### LINKED ISSUE #460: Update Crystallized State Once Per Cycle
Currently, [`setCrystallizedState`](https://github.com/prysmaticlabs/prysm/blob/f0abbf6f2679d9606e71b0c9d37113a7a2300d3a/beacon-chain/blockchain/service.go#L203) gets called every slot, even though it only changes once per cycle. Check that the state changed before writing to disk.

### PR DESCRIPTION
**Pull request by IvanTheGreatDev**
This fixes #460, not sure if there's more work needed.

### CODE REVIEW

**Comment by rauljordan** [comment]
Coverage here is pretty low, but aside from that LGTM

**Comment by IvanTheGreatDev** [comment]
Trying to figure out what's going on, coverage won't display for some reason. Do I add tests for the cases that should fail?

File: beacon-chain/blockchain/service.go [hunk 1]
```diff
@@ -217,6 +199,23 @@ func (c *ChainService) updateHead(slot uint64) {
 	// server to stream these events to beacon clients.
 	transition := c.chain.IsCycleTransition(slot)
 	if transition {
+		if err := c.chain.SetCrystallizedState(c.candidateCrystallizedState); err != nil {
+			log.Errorf("Write crystallized state to disk failed: %v", err)
+		}
+
+		// TODO: Utilize this value in the fork choice rule.
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/service.go]: Please reference a github issue. The TODO should be in the format TODO(1234) where 1234 is the issue number.

I'm not sure if there is a fork choice rule issue. @rauljordan  can you confirm?

**Comment by rawfalafel** [comment]
Hey, with the recent refactor this PR becomes simpler. I pushed a branch with my solution here: https://github.com/prysmaticlabs/prysm/compare/master...rawfalafel:update-crystallized-transition. Can you update this branch with the new code? Thanks.

**Comment by rauljordan** [comment]
Hey @IvanTheGreatDev do you have a chance to take a look at this this week?

**Comment by IvanTheGreatDev** [comment]
Yeah I do, I'll take a look at it soon.

**Comment by IvanTheGreatDev** [comment]
Apologies, did not mean to close this. I need some assistance with writing a test for the cycle transition. I tried to merge the latest changes into the branch and there are just too many, tests fails all over the place, so I just reset to latest and moved what needed to be moved into the cycle transition.

**Comment by IvanTheGreatDev** [comment]
Apologies, pushed the proper code now. Still need help with setting up a test.

**Comment by rauljordan** [comment]
Hey @IvanTheGreatDev, we ended up doing some refactoring of blockchain service in some latest PRs and this actually got resolved by a block validity fix we had to merge in in #541

**Pull request closed without merging**